### PR TITLE
Restructure/estimators

### PR DESCRIPTION
**Pull request by wkerzendorf**
An attempt at restructuring the estimator code. 

requires #2492 to be merged.

### CODE REVIEW

File: tardis/montecarlo/estimators/estimator_statistics.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    ("photo_ion_estimator", float64[:, :]),
+    ("stim_recomb_estimator", float64[:, :]),
+    ("bf_heating_estimator", float64[:, :]),
+    ("stim_recomb_cooling_estimator", float64[:, :]),
+    ("photo_ion_estimator_statistics", int64[:, :]),
+]
+
+
+@jitclass(base_estimators_spec + continuum_estimators_spec)
+class EstimatorStatistics:
```
Comment (wkerzendorf) [inline tardis/montecarlo/estimators/estimator_statistics.py:84]: RadiationFieldStatistics ?

Comment (wkerzendorf) [reply tardis/montecarlo/estimators/estimator_statistics.py:84]: `RadiationFieldStatisticalEstimators`

File: tardis/montecarlo/estimators/estimator.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+from tardis.montecarlo.estimators.util import bound_free_estimator_array2frame
+import tardis.constants as const
+
+H = const.h.cgs.value
+
+
+class ContinuumEstimators:
```
Comment (wkerzendorf) [inline tardis/montecarlo/estimators/estimator.py:7]: `EstimatedContinuumProperties`

File: tardis/montecarlo/estimators/estimator_statistics.py (hunk 1 above)
Comment (wkerzendorf) [reply tardis/montecarlo/estimators/estimator_statistics.py:84]: `RadiationFieldMCEstimators`

File: tardis/montecarlo/estimators/estimator.py (hunk 2 above)
Comment (andrewfullard) [reply tardis/montecarlo/estimators/estimator.py:7]: `ContinuumMCEstimators`

File: .hypothesis/unicode_data/11.0.0/charmap.json.gz
Comment (andrewfullard) [inline .hypothesis/unicode_data/11.0.0/charmap.json.gz:1]: What is this file doing here?

File: astropy_helpers
Comment (andrewfullard) [inline astropy_helpers:1]: Are we using astropy helpers now?

File: tardis.code-workspace [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
```
Comment (andrewfullard) [inline tardis.code-workspace:1]: Accidental commit?

File: tardis/model/radiation_field_state.py [hunk 4]
```diff
@@ -44,4 +46,6 @@ def calculate_mean_intensity(self, nu):
         intensity : u.Quantity
             Intensity of the radiation field at the given frequency
         """
-        return self.dilution_factor * intensity_black_body(nu, self.t_radiative)
+        return self.dilution_factor * intensity_black_body(
+            nu[np.newaxis].T, self.t_radiative
```
Comment (andrewfullard) [inline tardis/model/radiation_field_state.py:50]: Why is this transpose happening here?

Comment (wkerzendorf) [reply tardis/model/radiation_field_state.py:50]: this is the number of nues vs the number of shells - what comes out is a 2D frame

File: tardis/montecarlo/estimators/estimator.py (hunk 2 above)
Comment (wkerzendorf) [reply tardis/montecarlo/estimators/estimator.py]: Ah - In the end we decided for `ContinuumEstimators` as this is used by both MC and analytical methods

File: tardis/montecarlo/estimators/util.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    block_references : numpy.ndarray, dtype int
+        1D array with the start indices of the blocks to be integrated.
+
+    Returns
+    -------
+    numpy.ndarray, dtype float
+        2D array with integrated values.
+    """
+    integrated = np.zeros((len(block_references) - 1, f.shape[1]))
+    for i in prange(f.shape[1]):  # columns
```
Comment (andrewfullard) [inline tardis/montecarlo/estimators/util.py:62]: This doesn't respect the nthreads setting!

**Pull request closed without merging**